test/Tester/StreamingTests/PersistentStreamQueueRebalancingTestRunner.cs

```diff
@@ -0,0 +1,73 @@
+﻿
+using System;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+using Orleans;
+using Orleans.Hosting;
+using Orleans.Runtime.Development;
+using Orleans.TestingHost.Utils;
+using UnitTests.GrainInterfaces;
+
+namespace Tester.StreamingTests
+{
+    [TestCategory("Functional"), TestCategory("Streaming")]
+    public abstract class PersistentStreamQueueRebalancingTestRunner
+    {
+        private static TimeSpan Timeout = TimeSpan.FromMinutes(2);
+        private readonly ITestOutputHelper output;
+        private readonly IGrainFactory grainFactory;
+        private readonly string streamProviderName;
+
+        protected PersistentStreamQueueRebalancingTestRunner(string streamProviderName, IGrainFactory grainFactory, ITestOutputHelper output)
+        {
+            this.streamProviderName = streamProviderName;
+            this.output = output;
+            this.grainFactory = grainFactory;
+        }
+
+        public static ISiloHostBuilder ConfigureSiloHostBuilder(ISiloHostBuilder hostBuilder)
+        {
+            return hostBuilder.UseInMemoryLeaseProvider()
+                              .AddMemoryGrainStorage("PubSubStore");
+        }
+
+        [SkippableFact, TestCategory("Functional"), TestCategory("Streams")]
+        public async Task RebalanceQueuesWhileStreamingTest()
+        {
+            var streamGuid = Guid.NewGuid();
+
+            var producer = this.grainFactory.GetGrain<ISampleStreaming_ProducerGrain>(Guid.NewGuid());
+            await producer.BecomeProducer(streamGuid, streamGuid.ToString(), this.streamProviderName);
+
+            var consumer = this.grainFactory.GetGrain<ISampleStreaming_ConsumerGrain>(Guid.NewGuid());
+            await consumer.BecomeConsumer(streamGuid, streamGuid.ToString(), this.streamProviderName);
+
+            IDevelopmentLeaseProviderGrain leaseProviderGrain = InMemoryLeaseProvider.GetLeaseProviderGrain(this.grainFactory);
```

Comment:
I wonder do we need to spin up more consumer to test this. Maybe out of luck, the queue which is feeding this consumer never moved to another silo.